src/app/clusters/on-off-server/on-off-server.h

```diff
@@ -87,6 +87,12 @@ struct OnOffEffect
     OnOffEffect(
         chip::EndpointId endpoint, OffWithEffectTriggerCommand offWithEffectTrigger,
         uint8_t effectIdentifier = static_cast<uint8_t>(EMBER_ZCL_ON_OFF_EFFECT_IDENTIFIER_DELAYED_ALL_OFF),
+
+        /*
+         * effectVariant's type depends on the effect effectIdentifier so we don't know the type at compile time.
+         * Casting to uint8_t for more flexibilty since the type can be OnOffDelayedAllOffEffectVariant or
+         * OnOffDelayedAllOffEffectVariant
+         */
         uint8_t effectVariant = static_cast<uint8_t>(EMBER_ZCL_ON_OFF_DELAYED_ALL_OFF_EFFECT_VARIANT_FADE_TO_OFF_IN_0P8_SECONDS));
```

Comment:
Could probably handle this via chip::Variant...  Followup for that?

Reply:
fixed in #12813